File: conformance/reference-python/component_format.py
import math


def validate_component_format(input_value):
    format_value = input_value["format"]
    value = input_value.get("value")
    if format_value in ("currency", "percent"):
        if isinstance(value, bool) or not isinstance(value, (int, float)) or not math.isfinite(value):
            return {"ok": False, "code": "COMPONENT_DATA_TYPE_MISMATCH"}
    elif format_value == "datetime" and not isinstance(value, str):
        return {"ok": False, "code": "COMPONENT_DATA_TYPE_MISMATCH"}
    return {"ok": True, "value": value}

File: conformance/reference-js/component-format.js
'use strict';

function validateComponentFormat(input) {
  const { format, value } = input;
  if (format === 'currency' || format === 'percent') {
    if (typeof value !== 'number' || !Number.isFinite(value)) {
      return { ok: false, code: 'COMPONENT_DATA_TYPE_MISMATCH' };
    }
  } else if (format === 'datetime' && typeof value !== 'string') {
    return { ok: false, code: 'COMPONENT_DATA_TYPE_MISMATCH' };
  }
  return { ok: true, value };
}

module.exports = { validateComponentFormat };
